File: ezbuy-noti/src/main/java/com/ezbuy/noti/tasks/SendNotificationTasks.java
package com.ezbuy.noti.tasks;

import com.ezbuy.noti.service.TransmissionService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.scheduling.annotation.Async;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

@Component
@Slf4j
@RequiredArgsConstructor
public class SendNotificationTasks {

    private final TransmissionService transmissionService;

    @Async
    @Scheduled(fixedRate = 30000, initialDelay = 5000)
    public void sendNotificationTask() {
        log.info("=========Job send notification start==========");
        transmissionService.sendNotification().subscribe();
        log.info("==========Job send notification end==========");
    }
}

File: notification-send/src/main/kotlin/com/ezbuy/notificationsend/tasks/SendNotificationTasks.kt
package com.ezbuy.notificationsend.tasks

import com.ezbuy.notificationsend.service.TransmissionService
import org.slf4j.LoggerFactory
import org.springframework.scheduling.annotation.Async
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

@Component
class SendNotificationTasks(
    private val transmissionService: TransmissionService
) {

    private val log = LoggerFactory.getLogger(SendNotificationTasks::class.java)

    @Async
    @Scheduled(
        fixedRateString = "\${scheduled.fixedRate}",
        initialDelayString = "\${scheduled.initialDelay}"
    )
    fun sendNotificationTask() {
        val dateFormat = DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm:ss")
        log.info("=============================================")
        log.info("=====> STARTING JOB: SEND NOTIFICATION <=====")
        log.info("[START]\t\t Start: ${LocalDateTime.now().format(dateFormat)}")
        transmissionService.sendNotification()
            .doOnError { log.error("[ERROR] Error while sending notification: ${it.message}") }
            .doOnSuccess { log.info("[SUCCESS] Notification stream completed successfully") }
            .doFinally {
                log.info("=====> JOB FINISHED: SEND NOTIFICATION <=====")
                log.info("[END]\t\t   End: ${LocalDateTime.now().format(dateFormat)}")
                log.info("=============================================\n")
            }
            .subscribe()
    }
}